### PR TITLE
Skip flaky test_faulthandler.py::test_timeout[True] test on Windows

### LINKED ISSUE #7022: test_faulthandler.py::test_timeout[True] occasionally crashes
This tests the `faulthandler_timeout` functionality which dumps stacktraces of all threads of tests that run for more than a specified amount of times.

The test sometimes crashes with `Windows fatal exception: access violation` while dumping the threads. This is probably a cpython bug.

Noticed on both `windows-py35` and `windows-py37` jobs at least. (Update: also on `pypy3-ubuntu`).

For now I think we should just skip it on Windows, unless anyone has better ideas or ability to investigate. I'll send a PR doing that.

### PR DESCRIPTION
**Pull request by bluetech**
Fixes #7022.

### CODE REVIEW

File: testing/test_faulthandler.py [hunk 1]
```diff
@@ -71,8 +84,6 @@ def test_timeout():
 
     result = testdir.runpytest_subprocess(*args)
     tb_output = "most recent call first"
-    if sys.version_info[:2] == (3, 3):
```
Comment (bluetech) [inline testing/test_faulthandler.py:74]: (Also removed this, we don't support Python 3.3 so not needed anymore)

**Comment by blueyed** [comment]
It also just failed with pypy in https://github.com/pytest-dev/pytest/runs/562142405.

I think it might be worth trying to bump the timeout - although the behavior on/with timeout appears to be buggy at least on pypy (no real/full stacktrace).

File: testing/test_faulthandler.py [hunk 2]
```diff
@@ -49,8 +51,19 @@ def test_disabled():
...
 
-@pytest.mark.parametrize("enabled", [True, False])
-def test_timeout(testdir, enabled):
+@pytest.mark.parametrize(
+    "enabled",
+    [
+        pytest.param(
+            True,
+            marks=pytest.mark.skipif(
+                sys.platform.startswith("win"), reason="issue #7022",
```
Comment (blueyed) [inline testing/test_faulthandler.py:60]: Please provide a "real" reason.  Issue numbers have to be looked up etc.  You can sill have it in parenthesis there then.

**Comment by bluetech** [comment]
Not Windows specific, so closing.

**Pull request closed without merging**